apps/api-journeys/src/app/modules/block/form/form.resolver.ts

```diff
@@ -0,0 +1,155 @@
+export class FormBlockResolver {
+  constructor(
+    private readonly blockService: BlockService,
+    private readonly prismaService: PrismaService
+  ) {}
+
+  @Mutation()
+  @UseGuards(AppCaslGuard)
+  async formBlockCreate(
+    @CaslAbility() ability: AppAbility,
+    @Args('input') input: FormBlockCreateInput
+  ): Promise<Block> {
+    const parentOrder = (
+      await this.blockService.getSiblings(input.journeyId, input.parentBlockId)
+    ).length
+
+    return await this.prismaService.$transaction(async (tx) => {
+      const block = await tx.block.create({
+        data: {
+          ...omit(input, 'parentBlockId', 'journeyId'),
+          id: input.id ?? undefined,
+          typename: 'FormBlock',
+          journey: { connect: { id: input.journeyId } },
+          parentBlock: { connect: { id: input.parentBlockId } },
+          parentOrder
+        },
+        include: {
+          action: true,
+          journey: {
+            include: {
+              team: { include: { userTeams: true } },
+              userJourneys: true
+            }
+          }
+        }
+      })
+
+      if (!ability.can(Action.Update, subject('Journey', block.journey)))
+        throw new GraphQLError('user is not allowed to create block', {
+          extensions: { code: 'FORBIDDEN' }
+        })
+      return block
+    })
+  }
+
+  @Mutation()
+  @UseGuards(AppCaslGuard)
+  async formBlockUpdate(
+    @CaslAbility() ability: AppAbility,
+    @Args('id') id: string,
+    @Args('input') input: FormBlockUpdateInput
+  ): Promise<Block> {
+    const block = await this.prismaService.block.findUnique({
+      where: { id },
+      include: {
+        journey: {
+          include: {
+            team: { include: { userTeams: true } },
+            userJourneys: true
+          }
+        }
+      }
+    })
+    if (block == null)
+      throw new GraphQLError('block not found', {
+        extensions: { code: 'NOT_FOUND' }
+      })
+    if (!ability.can(Action.Update, subject('Journey', block.journey)))
+      throw new GraphQLError('user is not allowed to update block', {
+        extensions: { code: 'FORBIDDEN' }
+      })
+    return await this.blockService.update(id, input)
+  }
+
+  @ResolveField('projects')
+  async projects(@Parent() block: Block): Promise<FormiumProject[]> {
+    const { apiToken } = block
+    if (apiToken === null) return []
+
+    try {
+      const projectsData =
+        (await (await createClient('', { apiToken })).getMyProjects()).data ??
+        []
+
+      return projectsData.map((project) => ({
+        id: project.id,
+        name: project.name
+      }))
+    } catch (e) {
+      return []
+    }
+  }
+
+  @ResolveField('forms')
+  async forms(@Parent() block: Block): Promise<FormiumForm[]> {
+    const { projectId, apiToken } = block
+    if (projectId == null || apiToken == null) return []
+
+    try {
+      const formsData =
+        (
+          await (
+            await createClient(projectId, {
```

Comment:
```suggestion
            createClient(projectId, {
```

does not need to be awaited